src/shared/utils/api/getSubmissionInfo.ts

```diff
@@ -80,27 +87,32 @@ export const getSubmissionInfo = async (sid: string): Promise<ReturnType> => {
       };
     }
 
-    if (submissionInfo.runs.subs_code === null) {
+    if (!submissionInfo.code || !submissionInfo.error) {
       throw new Error(`Submission ${sid} has no subs_code`);
     }
 
+    if (!submissionInfo.name || !submissionInfo.tid) {
+      throw new Error(`Submission ${sid} has no teams.`);
+    }
```

Comment:
- Relation enforced at prisma level ensure this is not nullable
- Nullable can only caused by codefun, which should better throw Internal Server Error.
```suggestion
```